src/printer.js

```diff
@@ -2977,6 +2971,44 @@ function hasLeadingOwnLineComment(text, node) {
   return res;
 }
 
+// This recurses the return argument, looking for the first token
```

Comment:
note that with the code reorg, I figured this part of the comment was unnecessary: 

```
-// A "leadingComment" attached to the top of the first token in the
-// return's argument, which would originally be in parens,
-// could trigger an ASI error where the parens are removed
-// and the comment bumped to the same line as the `return`,
-// while the contents of the return value are on the next line (oops!).
-//
```
happy to add it in back in if you like